src/app/modules/shared/components/details-fields/details-fields.component.spec.ts

```diff
@@ -576,6 +576,12 @@ describe('DetailsFieldsComponent', () => {
     const foundActivity = component.findInactiveActivity(state.activities.data);
     expect(foundActivity).toEqual(expectedActivity);
   });
+
+  it('should return false when the start time entry is not greater than the end time', () => {
+    const result = component.isStartTimeEntryAfterEndedEntry();
+
+    expect(result).toBeFalse();
+  });
```

Comment:
In this part I think it we need more test cases to ensure that the output will not fail. I've prepared a table with the all combinations and outputs.

I would suggest to select the most relevant cases. 😄 

![image](https://user-images.githubusercontent.com/20387293/116875115-ff0c8780-abdf-11eb-80a4-f1bd374c1b53.png)

Reply:
For what @thegreatyamori  recommends you might consider using ` parameterized tests` similar to those found in line 538.
You could pass the values to the form using the patchValue something like this:

```
const FormDateValues = {
      start_date: '2021-05-15',
      end_date: '2021-04-14',
      start_hour: '15:05',
      end_hour: '17:00',
  };

    component.entryForm.patchValue(FormDateValues);
    const result = component.isStartTimeEntryAfterEndedEntry();
```